PIRK-45 Remove dependency on Java serialization

wraydulany opened the pull request:
This is a WIP (do not merge) for adding in JSON serialization. I can't get this WIP to pass the elasticsearch spark streaming tests on my cluster.

I'm fairly sure that this is a configuration problem on my end (the errors I get appear to be such), but I'm having little luck resolving them, and would love it if some of the community could give it a test.

I've left in Java Serialization for now, but I've made JSON serialization the default.

smarthi commented on src/main/java/org/apache/pirk/querier/wideskies/QuerierDeserializer.java:
```diff
@@ -0,0 +1,117 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.google.gson.*;
```
Please avoid using import *.

wraydulany commented on src/main/java/org/apache/pirk/query/wideskies/QueryDeserializer.java:
```diff
@@ -0,0 +1,265 @@
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.math.BigInteger;
+import java.util.*;
```
Yeah, there are 8 of those. Fixing.

wraydulany commented:
Fixed the `import...*` and also removed some code garbage that i'd commented out, from an earlier attempt to use jackson, an attempt killed by spark and its ancient version of jackson.

ellisonanne commented on src/main/java/org/apache/pirk/query/wideskies/Query.java:
```diff
@@ -26,100 +26,103 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Consumer;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
```
Need to remove Jackson imports

ellisonanne commented:
A couple of general comments - (1) need to remove the Jackson imports and dependency as it GSON is being used (2) need to format with the Pirk codestyle file.

I'll take a look at the ES issues...

wraydulany commented:
Ok, I've expurgated references (import and comment) to jackson in the code, and I've run the style checker against the code i touched. 

Additionally, I've investigated the problems with the 1:ES and 1:ESS tests that I'm having, and I'm convinced that they are due to misconfiguration of the cluster I'm testing on.

This passes the build tests and distributed tests 1:J, 1:JS, 1:JSS, 1:SS, and 1:E for me. If anyone has been able to successfully execute the 1:ES and/or 1:ESS tests with this PR *and* fail them, please comment and kill this PR. Otherwise, I'm going to move this from WIP to full PR in the late morning on the US East Coast tomorrow.

wraydulany commented:
Ok, changing to full PR. I should note for the record that the previously-working-for-me Pirk Releases also no longer pass the ES and ESS tests for me, another sign that It's Not My Fault.

ellisonanne commented:
+1 - looks good to merge

tellison commented on src/main/java/org/apache/pirk/query/wideskies/Query.java:
```diff
@@ -26,41 +26,51 @@
- *
  */
+
 public class Query implements Serializable, Storable
```
I assume we can also drop ```implements Serializable``` now too, right?
This instance, and friends elsewhere.

tellison commented on src/main/java/org/apache/pirk/query/wideskies/Query.java:
```diff
@@ -26,41 +26,51 @@
   // File based lookup table for modular exponentiation
   // element hash -> filename containing it's <power, element^power mod N^2> modular exponentiations
-  private Map<Integer,String> expFileBasedLookup = new HashMap<>();
+  @Expose private Map<Integer,String> expFileBasedLookup = new HashMap<>();
```
Should the hash -> filename map be serialized?
AFAIK it is only used by the responder to compute the results, so I don't think it needs serializing.

tellison commented on src/main/java/org/apache/pirk/query/wideskies/Query.java:
```diff
@@ -114,8 +124,7 @@ public void generateExpTable()
     {
-      @Override
-      public void accept(BigInteger element)
+      @Override public void accept(BigInteger element)
```
IDE formatting wars ;-)
